### PR TITLE
feat: use the Azure CLI as an auth method

### PR DESCRIPTION
**Pull request by bacongobbler**
Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

**Reason for Change**: This PR allows a user to authenticate using the Azure CLI 2.0 for development purposes. Marking this as a WIP as I haven't manually ran through with this; just experimenting with alternative auth methods.

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -151,7 +151,7 @@ ifndef HAS_GINKGO
 endif
 
 build-vendor:
-	${DEV_ENV_CMD} dep ensure
+	${DEV_ENV_CMD} dep ensure --vendor-only
```
Comment (tariq1890) [inline Makefile]: I think we'd want `--vendor-only`  to remain disabled for our current vendoring and vendor validations.

File: pkg/armhelpers/azureclient.go [hunk 2]
```diff
@@ -303,7 +325,7 @@ func getAcsEngineClientID(envName string) string {
 	}
 }
 
-func getClient(env azure.Environment, subscriptionID, tenantID string, armSpt *adal.ServicePrincipalToken, graphSpt *adal.ServicePrincipalToken) *AzureClient {
```
Comment (tariq1890) [inline pkg/armhelpers/azureclient.go]: Thanks for this :)

**Comment by bacongobbler** [comment]
I've gone through and manually tested this so I believe it's ready for review (once I rebase this).

To test, run `az login` as usual, then interact with `aks-engine` as usual. No service principal is required for authentication, but you'll still need to create one for the cluster itself:

```
$ az login
$ az group create -n $RG_NAME -l $LOCATION
$ az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/$SUBID/resourceGroups/$RG_NAME"
$ az deploy ...
```

**Comment by CecileRobertMichon** [comment]
unit tests failes `validate-dependencies`

**Comment by bacongobbler** [comment]
d'oh. Thanks for catching that. Will fix

**Comment by bacongobbler** [comment]
yes! I will go and rebase this now. Thanks for the heads up.

**Comment by bacongobbler** [comment]
to clarify: yes, this is ready for review.

**Pull request merged**